private/bufpkg/bufcheck/buflint/buflintconfig/buflintconfig.go

```diff
@@ -131,6 +132,22 @@ func ProtoForConfig(config *Config) *lintv1.Config {
 	}
 }
 
+// ValidateProtoConfig validates that the proto config is valid.
+func ValidateProtoConfig(protoConfig *lintv1.Config) error {
+	// An empty config is valid.
+	if protoConfig == nil {
+		return nil
+	}
+	version := protoConfig.GetVersion()
+	switch version {
+	case v1Version, v1Beta1Version:
+		return nil
+	default:
+		return fmt.Errorf("invalid lint config version %q provided", version)
+	}
```

Comment:
We need to validate a lot more than this. The Protobuf definition uses free-form strings for its IDs and `ignore_only` paths. At the very least, we should validate that each ID is real with respect to the configured version _and_ that every path is normalized.

Plus, I think this should exist on the server side in the module validator before anything else. We can export functions like this for the BSR to use, but this placement is susceptible to being used by the CLI later on (i.e. accidentally by another engineer) even if we don't mean it to.